File: LeetCode/Hash Table/top_k_frequent_elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& arr, int k) {
        unordered_map<int, int> um;
        int n = arr.size();
    for (int i = 0; i < n; i++) 
        um[arr[i]]++;  
    vector<int> freq[n + 1]; 
    for (int i = 0; i < n; i++) { 
        int f = um[arr[i]]; 
        if (f != -1) { 
            freq[f].push_back(arr[i]); 
            um[arr[i]] = -1; 
        } 
    }  
    int count = 0; 
    vector<int >ans;
    for (int i = n; i >= 0; i--) { 
        for (int x : freq[i]) { 
            ans.push_back(x);
            cout << x << " "; 
            count++; 
            if (count == k) 
                return ans; 
        } 
    } 
        return ans;
        
    }
};

File: LeetCode/Heaps/top_k_frequent_elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hash = {}
        for i in nums:
            if i in hash:
                hash[i]+=1
            else:
                hash[i] =1
        heap = []
        for num , freq in hash.items():
            heapq.heappush(heap,(freq,num))
            if len(heap)>k:
                heapq.heappop(heap)
        ans = []
        for freq,num in heap:
            ans.append(num)
        return ans
